modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/utils/SdkComponent.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.parser.java.utils;
+
+import org.mule.runtime.api.meta.MuleVersion;
+import org.mule.sdk.api.annotation.MinMuleVersion;
+
+/**
+ * Holds information about an SdkComponent, its {@link MinMuleVersion} and the reason for that value.
+ *
+ * @since 4.6
+ */
+public class SdkComponent {
```

Comment:
This class is very confusing:

* First I think the name is wrong. This is not modeling an `SdkComponent`, but the result of a version calculation.
* Also because reason seems to be something that is always assumed to be non-null, does this actually represent a result in which component MMV > extension MMV? 

I would rethink this completely